contracts/drafts/issuance/Issuance.sol

```diff
@@ -146,49 +177,37 @@ contract Issuance is Ownable, StateMachine {
     }
 
     /**
-     * @dev Function to call repeatedly from frontend, until all investors receive their tokens
+     * @dev Function to cancel all investments
      */
-    function sendToNextInvestor() public onlyOwner {
-        require(
-            currentState == "DISTRIBUTING",
-            "Cannot send tokens now."
-        );
-        if (nextInvestor >= investors.length) {
-            transition("LIVE");
-        } else {
-            issuanceToken.mint(
-                investors[nextInvestor],
-                investments[investors[nextInvestor]].div(issuePrice)
-            );
-            nextInvestor = nextInvestor.add(1);
-        }
-    }
-
-    /**
-     * @dev Function to cancel the investment of a certain investor
-     */
-    function cancelInvestment() public {
+    function cancelAllInvestments() public onlyOwner{
         require (
             currentState == "OPEN",
             "Cannot cancel now."
         );
-        acceptedToken.transfer(msg.sender, investments[msg.sender]);
-        emit InvestmentCancelled(msg.sender, investments[msg.sender]);
-        investments[msg.sender] = 0;
+        transition("REFUNDING");
     }
 
     /**
-     * @dev Function to cancel all investments and close the issuance
+     * @dev Function to call repeatedly from frontend, until all investors are refunded
      */
-    function cancelAllInvestments() public onlyOwner{
-        require (
-            currentState == "OPEN",
-            "Cannot cancel now."
+    function refundNextInvestor() public onlyOwner nonReentrant {
```

Comment:
You can do the same as for `withdraw`. If the issuance goes to a FAILED state then you can just let every investor come and execute a `refund` function that gives them their money back.